validator/client/beacon-api/duties.go

```diff
@@ -45,3 +252,45 @@ func (c beaconApiDutiesProvider) GetAttesterDuties(ctx context.Context, epoch ty
 
 	return attesterDuties.Data, nil
 }
+
+// GetProposerDuties retrieves the proposer duties for the given epoch
+func (c beaconApiDutiesProvider) GetProposerDuties(ctx context.Context, epoch types.Epoch) ([]*apimiddleware.ProposerDutyJson, error) {
+	proposerDuties := apimiddleware.ProposerDutiesResponseJson{}
+	if _, err := c.jsonRestHandler.GetRestJsonResponse(ctx, fmt.Sprintf("/eth/v1/validator/duties/proposer/%d", epoch), &proposerDuties); err != nil {
+		return nil, errors.Wrapf(err, "failed to query proposer duties for epoch `%d`", epoch)
+	}
+
+	for index, proposerDuty := range proposerDuties.Data {
+		if proposerDuty == nil {
+			return nil, errors.Errorf("proposer duty at index `%d` is nil", index)
+		}
+	}
+
+	return proposerDuties.Data, nil
+}
+
+// GetSyncDuties retrieves the sync committee duties for the given epoch and validatorIndices
+func (c beaconApiDutiesProvider) GetSyncDuties(ctx context.Context, epoch types.Epoch, validatorIndices []types.ValidatorIndex) ([]*apimiddleware.SyncCommitteeDuty, error) {
+	jsonValidatorIndices := make([]string, len(validatorIndices))
+	for index, validatorIndex := range validatorIndices {
+		jsonValidatorIndices[index] = strconv.FormatUint(uint64(validatorIndex), 10)
+	}
+
+	validatorIndicesBytes, err := json.Marshal(jsonValidatorIndices)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal validator indices")
+	}
+
+	syncDuties := apimiddleware.SyncCommitteeDutiesResponseJson{}
+	if _, err := c.jsonRestHandler.PostRestJson(ctx, fmt.Sprintf("/eth/v1/validator/duties/sync/%d", epoch), nil, bytes.NewBuffer(validatorIndicesBytes), &syncDuties); err != nil {
+		return nil, errors.Wrap(err, "failed to send POST data to REST endpoint")
+	}
+
+	for index, syncDuty := range syncDuties.Data {
```

Comment:
You may want to check if `syncDuties.Data == nil` first, like you do in `GetCommittees`.